src/main/java/Duke.java

```diff
@@ -19,29 +19,53 @@ public static void runUI(ArrayList<Task> tasks){
 
         displayWelcome();
         while(isContinue) {
-            taskName = scanner.nextLine();
-            command = taskName.split(" ");
-            switch (command[0]) {
+            taskName = scanner.nextLine();      //the whole user command
+            command = taskName.split(" "); //the different parts pf the command
+            switch (command[0]) {               //according to the first part of the command
                 case "bye":
                     displayQuit();
                     isContinue = false;
                     break;
                 case "list":
-                    list(tasks);
-                    break;
+                    try {
+                        list(tasks);
+                    } catch (RuntimeException exList) {
+                        System.out.println("    ____________________________________________________________");
+                        System.out.println("     ☹ OOPS!!! There is no task in your list.");
+                        System.out.println("    ____________________________________________________________");
+                    } finally {
+                        break;
+                    }
+
                 case "done":
-                    done(tasks, (int)command[1].charAt(0));
-                    break;
+                    try{
+                        done(tasks, Integer.parseInt(command[1])-1);
+                    } catch (IndexOutOfBoundsException exDone) {
+                        System.out.println("    ____________________________________________________________");
+                        System.out.println("     ☹ OOPS!!! There is no task "+Integer.parseInt(command[1])+" in the list.");
+                        System.out.println("    ____________________________________________________________");
+                    } finally {
+                        break;
+                    }
+
                 case "delete":
-                    deleteTask(tasks, (int)command[1].charAt(0));
-                    break;
+                    try{
+                        deleteTask(tasks, Integer.parseInt(command[1])-1);
+                    } catch (IndexOutOfBoundsException exDelete) {
+                        System.out.println("    ____________________________________________________________");
+                        System.out.println("     ☹ OOPS!!! There is no task "+Integer.parseInt(command[1])+" in the list.");
+                        System.out.println("    ____________________________________________________________");
+                    } finally {
+                        break;
+                    }
+
                 case "todo":
                     try {
-                        taskName = taskName.substring(5);
+                        taskName = taskName.trim().substring(5);         //it is very important to trim the space at the end
```

Comment:
Instead of hard coding the starting index of the substring, using String.length() may be more effective and modifiable if the specs were to change.